cql3/statements/delete_statement.cc

```diff
@@ -79,6 +84,60 @@ void delete_statement::add_update_for_key(mutation& m, const query::clustering_r
 
 namespace raw {
 
+namespace {
+
+using namespace restrictions;
+
+/// True iff parameters indicate a multi-column comparison.
+bool is_multi_column(size_t column_count, const term* t) {
+    return column_count > 1 || dynamic_cast<const tuples::value*>(t) || dynamic_cast<const tuples::marker*>(t)
+            || dynamic_cast<const tuples::delayed_value*>(t);
+}
+
+/// True iff expr bounds clustering key from both above and below OR it has no clustering-key bounds at all.
+/// See #6493.
+bool bounded_ck(const expression& expr) {
+    return std::visit(overloaded_functor{
+            [] (bool b) { return !b; },
+            [] (const binary_operator& oper) {
+                return *oper.op == operator_type::EQ; // Without EQ, one side must be unbounded.
+            },
+            [] (const conjunction& conj) {
+                using bounds_bitvector = int; // Combined using binary OR.
+                static constexpr bounds_bitvector UPPER=1, LOWER=2;
+                std::unordered_map<const column_definition*, bounds_bitvector> found_bounds;
```

Comment:
I think it is nicer to move found_bounds outside the std::visit() call. This way, we remove the dependency on having just a single conjunction, and a tree of conjunctions works just as well.

This removes the duplication in the binary_operator overload - we just call the visitor recursively, and the inner visitor below disappears.

Reply:
Oh, and since we do have just a single conjunction, it can be a follow-up.